### DISCUSSION TITLE
StoreKit2: It seems that the Sk1 module is also listening to purchases when using STOREKIT2_MODE

### DISCUSSION THREAD

**Discussion by zarv1k** [hyochan/react-native-iap#2295] [General]
Hi there!

Using `STOREKIT2_MODE` I noticed some debug logs like the following:

```
[react-native-iap] Purchase Successful
[react-native-iap] requestReceiptDataWithBlock with force refresh: NO
```

So it seems that Sk1 module is still in use even the `STOREKIT2_MODE` is enabled.

Is it expected behaviour? As for me no, It's not normal. I'd like to expect that I'm using the Sk2 mode only. May be I'm wrong but I think it happens because there is a call of addTransactionObserver() at the constructor of RNIapIos module (`RNIapIos.init()`).

Any thoughts on that?

@andresesfm

**Comment by andresesfm** [comment]
Looks like a bug @zarv1k
